test-project/src/test_contract_tests.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Mutex;
+
+use aleph_client::{pallets::balances::BalanceUserApi, KeyPair, SignedConnection, TxStatus};
+use anyhow::Result;
+use assert2::assert;
+use ink_primitives::AccountId;
+use once_cell::sync::Lazy;
+use rand::RngCore as _;
+use test_contract::{Enum1, Struct1, Struct2};
+
+use crate::test_contract;
+
+static AUTHORITY_MUTEX: Lazy<Mutex<()>> = Lazy::new(|| Mutex::new(()));
+
+async fn connect_and_deploy() -> Result<(SignedConnection, test_contract::Instance)> {
+    let guard = AUTHORITY_MUTEX.lock().unwrap();
```

Comment:
Do we need the `Mutex` so that we don't use same nonce more than once?

Reply:
Ah, I forgot, meant to refactor this. Yes, the mutex is for that, but I think it would be clearer to just put the alice connection in the `Lazy`, instead of a `Mutex<()>`, WDYT?

Reply:
Ah, get rid of `test_account` entirely and init `Alice` once and reuse? That's fine too.

Reply:
No, no, we need the test account, see my other comment.